### ISSUE TITLE
reader.go:35: constant 4294967295 overflows int

### ISSUE THREAD

**Issue by brunoqc** [ulikunitz/xz#9]
```
# github.com/ulikunitz/xz/lzma
..\github.com\ulikunitz\xz\lzma\reader.go:35: constant 4294967295 overflows int
..\github.com\ulikunitz\xz\lzma\reader2.go:31: constant 4294967295 overflows int
..\github.com\ulikunitz\xz\lzma\writer.go:78: constant 4294967295 overflows int
..\github.com\ulikunitz\xz\lzma\writer2.go:55: constant 4294967295 overflows int
```

go 1.6.3
windows xp

**Comment by ulikunitz** [comment]
Many thanks for reporting the issue. It is caused by 32-bit integers on the 386 platform. It is fixed by commit d69e41a .

**Comment by ulikunitz** [comment]
Release v0.5.1 contains the fix.
